bin/desi_inspect_dark

```diff
@@ -0,0 +1,184 @@
+                    help = 'specify psf file for trace coordinates, default is automatically found.')
+parser.add_argument('--dist', type = float , default = 3., required = False,
+                    help = 'min distance in pixels between fiber trace and bad column')
+parser.add_argument('--nopsf', action = 'store_true',
+                    help = 'do not read the traces and do not associate bad columns to fibers')
+
+args        = parser.parse_args()
+
+
+log = get_logger()
+
+log.info("Reading "+args.infile)
+pix  = fitsio.read(args.infile)
+head = fitsio.read_header(args.infile)
+mask = fitsio.read(args.infile,"MASK")
+
+if not args.nopsf :
+    if args.psf is None :
+        args.psf = findcalibfile([head],"PSF")
+    log.info("Will read traceset in "+args.psf)
+
+    tset = read_xytraceset(args.psf)
+    fmap = read_fibermap(args.infile)
+    fibers = fmap["FIBER"]
+else :
+    tset   = None
+    fibers = None
+
+n0=pix.shape[0]
+n1=pix.shape[1]
+
+profs=[]
+mprofs=[]
+
+col_x=[]
+col_val=[]
+col_begin=[]
+col_end=[]
+
+for half in range(2) :
+    if half==0 :
+        b=0
+        e=n0//2
+    else :
+        b=n0//2
+        e=n0
+
+    # remove possible background before detecting bad columns
+    prof = np.zeros(n1)
+    mprof = np.zeros(n1)
+    good=np.where(np.sum(mask[b:e]==0,axis=0)>0)[0]
+    prof[good] = np.ma.median(np.ma.masked_array(data=pix[b:e,good],mask=(mask[b:e,good]!=0)),axis=0).data
+
+    step=30 # must be more than twice as large as bad column, but not to wide to follow variations
+    for lr in range(2) :
+        if lr==0 :
+            pb=0
+            pe=n1//2
+        else :
+            pb=n1//2
+            pe=n1
+        medf = scipy.ndimage.median_filter(prof[pb:pe],step)
+        mmedf = np.max(np.abs(medf))
+        if mmedf> 0.5 :
+            log.warning("anomalous residual background in dark of {:.2f} electrons".format(mmedf))
+        prof[pb:pe] -= medf
+        mprof[pb:pe] = medf
+
+    profs.append(prof)
+    mprofs.append(mprof)
+
+    # detect cols
+    aprof = np.abs(prof)
+    peak = (aprof>args.threshold/2) # half of threshold to select columns for further inspection
+    peak[1:-1] &= (aprof[1:-1]>aprof[:-2])&(aprof[1:-1]>aprof[2:])
+    peak=np.where(peak)[0]
+
+    for p in peak :
+        # median filter for column that do not cross the whole amplifier
+        medval = scipy.ndimage.median_filter(pix[b:e,p],400)
+        i = np.argmax(np.abs(medval))
+        val = medval[i]
+        # apply threshold here
+        if np.abs(val)<args.threshold :
+            continue
+        log.info("Bad column x={} val={:.2f}".format(p,val))
+        col_x.append(p)
+        col_val.append(val)
+
+
+if not args.nopsf : # add fiber info
+    wrange = (tset.wavemax-tset.wavemin)
+    wmean  = (tset.wavemin+tset.wavemax)/2.
+    wave   = np.array([wmean-wrange*0.4,wmean, wmean+wrange*0.4]) # do not test the whole range because on signal on edges
```

Comment:
This should sample more wavelengths.  e.g. night 20201219 r1 identified column 3898 as bad and flagged fibers 988 and 990 as bad but not fiber 989.  Looking at the full xytraceset, fiber 989 is the only one that actually overlaps the column:
![image](https://user-images.githubusercontent.com/218471/130112438-c4d7d5f8-345a-4106-a3d5-c9bf50134b5d.png)
the colored regions are the portion of each trace within the default 3 pixel proximity cut.